pkg/interactive/docker.go

```diff
@@ -76,6 +78,52 @@ func (d *Docker) AddMount(typeStr, source, target string, optAdditionalArgs ...s
 	d.AddArgs("--mount", fmt.Sprintf("type=%s,source=%s,target=%s%s", typeStr, source, target, addl))
 }
 
+// CopyAndAddMount copies the source files into a temp directory, and then
+// mounts them as the target. It also returns a function to remove the temp
+// directory for cleaning up.
+func (d *Docker) CopyAndAddMount(typeStr, parentDir, source, target string, optAdditionalArgs ...string) func() {
+	fi, err := os.Stat(source)
+	if err != nil {
+		log.Fatalf("Error getting the FileInfo for %q: %v", source, err)
+		return nil
+	}
+
+	tempDir, err := ioutil.TempDir(parentDir, fi.Name())
+	if err != nil {
+		log.Fatal("error creating a temporary directory: ", err)
+		return nil
+	}
+
+	cleanup := func() {
+		log.Printf("🧹 Removing the temp directory %q", tempDir)
+		if err := os.RemoveAll(tempDir); err != nil {
+			log.Printf("Warning: error removing the temp directory %q: %v", tempDir, err)
+		}
+	}
+
+	if err := os.Chmod(tempDir, 0777); err != nil {
+		cleanup()
+		log.Fatalf("error changing file mode for %q: %v", tempDir, err)
+	}
+
+	// Copy all the content if it's a directory
+	if fi.IsDir() {
+		source += "/."
+	}
+	if _, err := cmd.RunCommand(fmt.Sprintf("/bin/bash -c 'cp -r %s %s'", source, tempDir), cmd.WithStdout()); err != nil {
```

Comment:
why bash? you should be able to run `cp` directly

Reply:
The reason was I used to run as `cp -r xxx/* yyy`, and `*` can only be used with bash. But now since I'm not using `*` anymore, I'll remove the bash part then.